lib/middleware.js

```diff
@@ -1,10 +1,61 @@
 'use strict';
 
+require('dotenv').load();
+const jwt = require('jwt-simple');
+
 module.exports = function(req, res, cb) {
-  /* To be replaced with token validation and/or authentication */
-  const validated = Object.assign(req, {
-    user: 'test-user',
-    isAdmin: false
-  });
-  return cb(null, validated);
+  const token = getToken(req);
+  let payload;
+  if (token) {
+    payload = getPayload(token);
+  } else {
+    payload = null;
+  }
+
+  // QUESTION: should we check if the user if belongs to the list
+  // of authorized users here and add like an `.isTrustedUser` property or
+  // so to the request?
+  if (payload) {
+    req.user = payload;
+    if (process.env.trustedUsers) {
+      const trustedUsers = JSON.parse(process.env.trustedUsers);
+      if (trustedUsers.indexOf(payload.id) !== -1) {
+        req.isTrustedUser = true;
+      } else {
+        req.isTrustedUser = false;
+      }
+    } else {
+      req.isTrustedUser = true; // if no env var, all OSM users are trusted
+    }
+  } else {
+    req.user = null;
+    req.isTrustedUser = false;
+  }
+
+  return cb();
```

Comment:
Just to confirm, this `req` object that now has `user` and `isTrustedUser` properties will be passed down to all downstream endpoints when we `return cb();`, or do we need to pass back the `req` object somehow?

Reply:
Based on my understanding of `express` middleware, yes, anything you add to the request object in middleware would then be available to further consumers downstream / those that are `next` in the cycle.